src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -95,14 +102,13 @@ public String getPattern() {
             listener.getLogger().format("Must specify user or organization%n");
             return;
         }
-        String apiUrl = null; // TODO GHE
-        StandardCredentials credentials = Connector.lookupScanCredentials(observer.getContext(), apiUrl, scanCredentialsId);
+        StandardCredentials credentials = Connector.lookupScanCredentials(observer.getContext(), null /* TODO apiUri */, scanCredentialsId);
```

Comment:
> As far as a I know, this will filter for this domain.

Exactly. That is what you want. Otherwise we are searching for credentials from unrelated domains. You are always free to define credentials in the global domain, in which case they will be visible unconditionally. CC @stephenc